File: frontend/src/app/models/lan.ts
export interface LanStatsRequest {
  seconds: number;
  numberOfReadings: number;
  interfaceNames: string[];
  beginTimestamp?: number;
  endTimestamp?: number;
}

export interface DeviceStatsRequest {
  seconds: number;
  numberOfReadings: number;
  deviceName: string;
  beginTimestamp?: number;
  endTimestamp?: number;
}

export interface MonitoringTestRequest {
  duration: number;
  interval: number;
}

File: backend/models/lan.py
from pydantic import BaseModel


class LanStatsRequest(BaseModel):
    Seconds: int
    NumberOfReadings: int
    InterfaceName: list[str]
    BeginTrafficTimestamp: int | None = None
    EndTrafficTimestamp: int | None = None


class DeviceStatsRequest(BaseModel):
    Seconds: int
    NumberOfReadings: int
    DeviceName: str
    BeginTrafficTimestamp: int | None = None
    EndTrafficTimestamp: int | None = None


class MonitoringTestRequest(BaseModel):
    duration: int
    interval: int
